### PR TITLE
YAML schemas

### PR DESCRIPTION
**Pull request by jtwb**
This PR completes work on the YAML schemas. Schemas are provided for config files (e.g. aerleon.yml), definition files (e.g. naming.yaml) and policy files.

These schemas have been validated using the AJV validator through schemastore's test suite:

```
$ npm run remote

> schemastore.org@1.0.0 remote
> grunt remote_test

Running "remote_assert_schema_no_bom" task

Running "remote_test_ajv" task

>> pass schema          | http://172.23.0.3:8000/schemas/aerleon-config.schema.json (draft-04)(FullStrictMode)
>> http://172.23.0.3:8000/schemas/aerleon-config.schema.json

>> pass schema          | http://172.23.0.3:8000/schemas/aerleon-definitions.schema.json (draft-07)(FullStrictMode)
>> http://172.23.0.3:8000/schemas/aerleon-definitions.schema.json

>> pass schema          | http://172.23.0.3:8000/schemas/aerleon-policies.schema.json (draft-07)(FullStrictMode)
>> http://172.23.0.3:8000/schemas/aerleon-policies.schema.json

Total schemas validated with AJV: 3
```

To test these changes in VSCode, perform the following steps. Note that once the schema is provided to json-schema-store, the RedHat YAML extension will no longer be required to use these schemas.

* Check out this branch.
* Install the RedHat YAML Extension: https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml
* Open the Settings panel in VSCode, search for "Schema", and edit your settings.json with the following config:
```
    "yaml.schemas": {
    
        "schemas/aerleon-config.schema.json": ["aerleon.yml"],
        "schemas/aerleon-policies.schema.json": ["*.policy.yaml", "policies/**/*.yaml"],
        "schemas/aerleon-definitions.schema.json": ["*.definition.yaml", "def/**/*.yaml"],
    },
```
* Open the aerleon.yml file to test the config schema.
* Open any policy YAML file to test the policy schema.
* Open the naming.yaml file to test the definitions schema.

**Future Work**

This PR does not attempt to document or validate any specific term keyword. A future version of this schema could provide detailed text for each built-in term keyword. Validating each term value may not be a good idea though, as we want to reserve the right for generator authors to extend the configuration language.

### CODE REVIEW

File: schemas/aerleon-config.schema.json [hunk 1]
```diff
@@ -1,52 +1,51 @@
 {
-  "$id": "urn:aerleon:schemas:aerleon-config:1.0.0",
   "$schema": "http://json-schema.org/draft-04/schema#",
   "title": "aerleon.yml",
-  "description": "Used to configure Aerleon's aclgen, aclcheck, and cgrep command line tools.",
+  "description": "configure Aerleon's aclgen, aclcheck, and pol2yaml command line tools",
```
Comment (ankenyr) [inline schemas/aerleon-config.schema.json:4]: Some descriptions start with upper case and others lowercase. Same for punctuation. Lets fix that?

Comment (jtwb) [reply schemas/aerleon-config.schema.json:4]: Better yet, some of these top-level descriptions can be removed.

File: schemas/aerleon-definitions.schema.json [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+      "title": "Network Definition Section",
+      "description": "Policy files can reference the networks in this section by name. A network is a list of IP addresses or CIDR IP address ranges and may contain other networks.",
       "type": "object",
       "additionalProperties": { "$ref": "#/$defs/networkDefinition" },
       "propertyNames": { "$ref": "#/$defs/token" }
     },
     "services": {
-      "description": "An Aerleon definition file defines named networks and services which can be referenced by Aerleon Policy files. Networks are lists of IP addresses or CIDR IP address ranges. Services are lists of port/protocol pairs (e.g. tcp/80) and can include port ranges.\nDefinition files must include a \"networks\" section, a \"services\" section, or both.\nNetworks and services may use composition in their definitions. Include another definition by creating an item containing the \"name\" property.",
+      "title": "Service Definition Section",
+      "description": "Policy files can reference the services in this section by name. A service is a list of port/protocol pairs (e.g. tcp/80) and can include port ranges. A service can include other services.",
```
Comment (ankenyr) [inline schemas/aerleon-definitions.schema.json:18]: The example here is backwards both in terms of how you refer to it `port/protocol` and how it is used `HTTP:: 80/tcp`. You should consider switching it in the example here to `80/tcp`
You may also wish to include a port range example.

Comment (ankenyr) [reply schemas/aerleon-definitions.schema.json:18]: You can axe the second part. Just realized this is only for the YAML spec so the `HTTP:: 80/tcp` makes no sense.

File: schemas/aerleon-definitions.schema.json [hunk 3]
```diff
@@ -42,10 +62,11 @@
           "description": "Defines a named network, composed of IP addresses, IP address ranges, and references to other networks.\nAn object with the \"address\" property adds that address or address range to the network.\nAn object with the \"name\" property includes the contents of that network into this one.\nA single string also includes the content of that network into this one.",
           "type": "array",
           "items": {
-            "description": "Defines a named network, composed of IP addresses, IP address ranges, and references to other networks.\nAn object with the \"address\" property adds that address or address range to the network.\nAn object with the \"name\" property includes the contents of that network into this one.\nA single string also includes the content of that network into this one.",
             "oneOf": [
               {
                 "type": "string",
+                "title": "Network Reference",
+                "description": "Include another network by name. The referenced network must be defined in this file.",
```
Comment (ankenyr) [inline schemas/aerleon-definitions.schema.json:68]: I don't believe this is accurate. You can define a network in one file and reference it in another.

File: schemas/aerleon-definitions.schema.json [hunk 4]
```diff
@@ -59,9 +80,13 @@
               },
               {
                 "type": "object",
+                "title": "Network Reference",
                 "required": ["name"],
                 "properties": {
-                  "name": { "$ref": "#/$defs/token" },
+                  "name": {
+                    "description": "Include another network by name. The referenced network must be defined in this file.",
```
Comment (ankenyr) [inline schemas/aerleon-definitions.schema.json:86]: Same here

File: schemas/aerleon-definitions.schema.json [hunk 5]
```diff
@@ -73,37 +98,31 @@
...
+              "comment": { "$ref": "#/$defs/comment" }
             },
             "additionalProperties": false
           },
           {
             "type": "object",
-            "title": "ServiceReference",
-            "description": "References another service.",
+            "title": "Service Reference",
+            "description": "Include another service by name. The referenced service must be defined in this file.",
```
Comment (ankenyr) [inline schemas/aerleon-definitions.schema.json:117]: Same

File: schemas/aerleon-definitions.schema.json [hunk 6]
```diff
@@ -73,37 +98,31 @@
...
             "required": ["name"],
             "properties": {
-              "name": { "$ref": "#/$defs/token" },
-              "comment": { "type": "string" }
+              "name": {
+                "description": "Include another service by name. The referenced service must be defined in this file.",
```
Comment (ankenyr) [inline schemas/aerleon-definitions.schema.json:121]: Same

File: schemas/aerleon-definitions.schema.json (hunk 3 above)
Comment (jtwb) [reply schemas/aerleon-definitions.schema.json]: Yeah, just confirmed this works fine.

**Review by ankenyr — APPROVED** [review]

**Pull request merged**